### ISSUE TITLE
[BUG]: Images in Image Occlusion cards are too small on AnkiDroid [landscape mode]

### ISSUE THREAD

**Issue by aaaaaad333** [ankidroid/Anki-Android#15235]
### Checked for duplicates?

- [X] This issue is not a duplicate

### What are the steps to reproduce this bug?

Create an Image Occlusion card in Anki desktop (Version ⁨23.12.1).
Sync and review it in AnkiDroid.

### Expected behaviour

Images fit to width.

### Actual behaviour

Images are very small and have to be zoomed in manually.

This is a card made with Anki's native Image Occlusion:

![test1](https://github.com/ankidroid/Anki-Android/assets/123825694/dfe1cd72-f19b-4280-b35c-9b7acc07d192)

This was made using the Image Occlusion Enhanced addon:

![test2](https://github.com/ankidroid/Anki-Android/assets/123825694/6b4f2090-5d94-4ff1-8a52-54767bfeca6a)

### Debug info

```text
AnkiDroid Version = 2.17alpha15 (e5b25f3fef737140ae396d1b212a8d638551500f)

Backend Version = 0.1.34-anki23.12.1 (23.12.1 1a1d4d5419c6b57ef3baf99c9d2d9cf85d36ae0a)

Android Version = 12 (SDK 31)

ProductFlavor = full

Manufacturer = HMD Global

Model = Nokia 5.4

Hardware = qcom

Webview User Agent = Mozilla/5.0 (Linux; Android 12; Nokia 5.4 Build/SKQ1.220119.001; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/120.0.6099.193 Mobile Safari/537.36

ACRA UUID = e3cee8ce-3afa-4813-a03d-d23e62bc8a76

Crash Reports Enabled = false
```

### (Optional) Anything else you want to share?

Update: the issue is still present in AnkiDroid 2.17alpha16

### Research

- [X] I am reporting a bug specific to AnkiDroid (Android app)
- [X] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [X] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

**Comment by user1823** [comment]
You can suggest this on Anki Forums (https://forums.ankiweb.net) so that they can fix it for everyone.

**Comment by user1823** [comment]
Here is what Damien has to say on this issue:

> The styling for each client will be different, so I think this should be styled by each client separately instead of done in the core code. AnkiMobile already injects its own styling so that the image appears correctly for it.

**Comment by david-allison** [comment]
> Here is what Damien has to say on this issue:
> 
> > The styling for each client will be different, so I think this should be styled by each client separately instead of done in the core code. AnkiMobile already injects its own styling so that the image appears correctly for it.

Thanks! Good first issue to apply the CSS provided above

**Comment by dennisko762** [comment]
Hey, Im currently also looking into this and for some reason Im either unable to reproduce it or to stupid. Maybe you can help me out.
Details: Im using the 2.17alpha16 on a samsung galaxy s20.
For me the images are fit nicely to the screen.

**Comment by user1823** [comment]
To reproduce this, the phone needs to be in the landscape mode. Probably, you are trying to reproduce it in the Portrait mode.

**Comment by dennisko762** [comment]
![Screenshot_20240210-142135_AnkiDroid](https://github.com/ankidroid/Anki-Android/assets/56516613/c021c5ea-cd6d-4b3c-80f8-d203fa2bafe3)

So this one is in landscape mode. Id say is rather big than too small. Or does the problem only occur with the native occlusion?

**Comment by user1823** [comment]
> Or does the problem only occur with the native occlusion?

Yes. As mentioned in the original post, it occurs only with the native IO cards.

> Id say is rather big than too small.

Probably, we should find a middle way between the current size in native IO cards and the size in add-on generared IO cards.

**Comment by QuiteChridi** [comment]
Hi, I was starting to look into this issue and was able to reproduce it. I tried adding the afore mentioned css styling to the "flashcard.css" file to fix it, but unfortunately it does not seem to solve the problem, it just shifts the image to the left without changing the size.

Am I missing something obvious here about the place where the android specific styling should happen?

**Comment by RedLexa** [comment]
I just reproduced this, and am looking into this!

**Comment by RedLexa** [comment]
I'm looking into this as my first issue! I've already replicated it think I have an idea of things to try on Anki Android to fix this issue, seen as tough it only seems to happen to me with certain image sizes. Could I have this assigned to me?

**Comment by RedLexa** [comment]
Hello! I'd like to give some updates regarding this issue, seen as though I've already fixed the main bug but am now facing another which I'm struggling to consider how to fix.

The CSS provided earlier in this thread works! But only if you !important the #image-occlusion-container img properties. However, a downside I wasn't expecting was that the occlusion itself now seems to be deformed.
![FIxed com Occlusion estragadad](https://github.com/ankidroid/Anki-Android/assets/109145734/14b02f7e-1778-4e75-93f4-fd0573f2cf1c)
The div containing the Occlusion doesn't have any id so I was not sure how to proceed.
`<div style="display: none">{{cloze:Occlusion}}</div>`
 Should I add an id to that div, so that I can style it along with other #image-occlusion-container img class, ensuring it stays in place?

**Comment by John4650-hub** [comment]
I have an idea to solve this but i can't implement it, i don't know whether it's possible but i think adding a floating button that a user can click to lock the  zooming  so that when going over the cards of type IO, the zooming remains consistent and the user can then decide when to change the zooming by clicking that lock button again so that they adjust the  zooming as required.
Iam kinda facing an issue similar to this. When i zoom-in on the diagram and then reveal the hidden parts, the zooming always restores itself to it's default, and when i try to zoom in , sometimes i fail to recognize which word or part of the image was covered.
